### PR TITLE
esptool: update deps and homepage

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/esptool.rb [hunk 1]
```diff
@@ -17,35 +17,33 @@ class Esptool < Formula
...
+  depends_on "openssl@1.1"
+  depends_on "pycparser"
   depends_on "python@3.11"
   depends_on "six"
 
-  resource "bitstring" do
-    url "https://files.pythonhosted.org/packages/d2/64/e733b18349be383a4b7859c865d6c9e5ccc5845e9b4258504055607ec1cb/bitstring-4.0.1.tar.gz"
-    sha256 "7719f08f6df89ce28453a5e580d4a8ec1d1bda892dbb033466da0ccd9bdcb706"
+  on_linux do
+    depends_on "pkg-config" => :build
```
Comment (carlocab) [inline Formula/esptool.rb:29]: Skip the `on_linux` block; `pkg-config` will already be installed because of `rust`.

Comment (chenrui333) [reply Formula/esptool.rb]: 👍

Comment (chenrui333) [reply Formula/esptool.rb]: as you can see, it does need `pkg-config`, https://github.com/Homebrew/homebrew-core/actions/runs/5108178647/jobs/9181831385?pr=132216

Comment (carlocab) [reply Formula/esptool.rb]: Yes, I meant list it along with the other dependencies instead of separating it in an `on_linux` block.

Comment (chenrui333) [reply Formula/esptool.rb]: I see

**Review by carlocab — APPROVED** [review]

**Review by carlocab — APPROVED** [review]

**Pull request merged**